src/Microsoft.Health.Api/Features/Audit/AuditEventTypeMapping.cs

```diff
@@ -70,7 +76,8 @@ public Task StartAsync(CancellationToken cancellationToken)
                 .Select(ad =>
                 {
                     Attribute attribute = ad.MethodInfo?.GetCustomAttributes<AllowAnonymousAttribute>().FirstOrDefault() ??
```

Comment:
As FhirAnonymoussOperationAttribute is derived from AnonymousAttribute, will the first condition in this chain return nonnull?

I am also curious why the AuditEventTypeAttribute has the (Attribute) cast, but it is unnecessary elsewhere?

Reply:
yes, it does return nonnull value. or in our specific case it returns metadata/versions.
For (Attribute) cast, I am also not so sure butI think that's because of ?? operator - it requires both the sides to be of same type. When we have FhirAnonymousOperationAttribute (which is derived from AllowAnonymousAttribute) and AllowAnonymousAttribute -> it doesn't require any casting may be because one is derived form the other. But when we compare AuditEventTypeAttribute with remaining two it does require casting.